app/windows/winget/winget-create.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * Usage:
+ *
+ * 1. Ensure that Node.js and npm are installed on your system.
+ * 2. Run this script using the command `node winget-create.mjs <projectVersion> <directoryPath>`.
+ *
+ * Parameters:
+ * - projectVersion: The version of the project you want to fetch information for (e.g., 0.19.1).
+ * - directoryPath: The path to the directory where the generated YAML files should be stored (e.g., app/windows/winget/manifests).
+ *
+ * Ensure that you replace `<projectVersion>` with the actual project version, do not include the 'v'.
+ *
+ * Note:
+ * - The templates directory should contain the necessary YAML templates and should be located in the same directory as this script.
+ * - The manifests directory will be created if it does not exist, and it will contain the generated YAML files.
+ */
+
+const fetch = require('node-fetch');
+const fs = require('fs');
+const path = require('path');
+
+const args = process.argv.slice(2);
+
+if (!(args.length === 2)) {
+  console.error('Usage: node winget-create.mjs <projectVersion> <directoryPath>');
+  console.error('Please provide the project version without the "v" prefix.');
+  console.error('Example: node winget-create.js 0.19.1 my/path/to/folder');
+  process.exit(1);
+}
+
+const projectVersion = args[0];
+
+const directoryPath = args[1];
+
+if (projectVersion.startsWith('v')) {
+  console.error('Please provide the project version without the "v" prefix.');
+  console.error('Example: node winget-create.js 0.19.1 my/path/to/folder');
+  process.exit(1);
+}
+
+let sha256;
+
+const arch = {
+  x64: 'x64',
+  x86: 'x86',
+};
+
+async function createYAMLFiles(version, downloadURL, date, notesURL) {
+  const templates = [
+    'Headlamp.Headlamp.installer.yaml',
+    'Headlamp.Headlamp.locale.en-US.yaml',
+    'Headlamp.Headlamp.yaml',
+  ];
+
+  // use path to get the template directory
```

Comment:
This comment is not adding much.

Reply:
I will remove all the notes I left in while learning how to make the script work from the original, they're basically to do notes anyway.